### PR TITLE
Add roomDestroyed to MUC UserStatusListener

### PR DESCRIPTION
**Pull request by meisterfuu**
SMACK-619

New PR(and fork) because work computer messed with the git usernames....

### CODE REVIEW

File: smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChat.java [hunk 1]
```diff
@@ -2179,6 +2179,18 @@ private void checkPresenceCode(
                 listener.nicknameChanged(from, mucUser.getItem().getNick());
             }
         }
+        //The room has been destroyed
+        if (mucUser.getDestroy() != null) {
+        	MultiUserChat alternateMUC = multiUserChatManager.getMultiUserChat(mucUser.getDestroy().getJid())
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChat.java]: looks like a whitespace issue here

File: smack-extensions/src/main/java/org/jivesoftware/smackx/muc/UserStatusListener.java [hunk 2]
```diff
@@ -123,4 +123,12 @@
      */
     public abstract void adminRevoked();
 
+    /**
+     * Called when the room is destroyed.
+     * 
+     * @param alternateMUC an alternate MultiUserChat.
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/muc/UserStatusListener.java]: IIRC alternateMUC and reason are both optional. If so, it should be mentioned in the javadoc that they may be null.

**Comment by meisterfuu** [comment]
The coverage decrease seems kinda random to me?!

**Comment by Flowdalic** [comment]
Well, you didn't write a unit test for the `mucUser.getDestroy() != null` condition, so when you add code that isn't covered by the unit test, the overall test coverage decreases. But although it would be nice to have a unit test for that case, not having one is not a show stopper.

What's your JIRA username (so I can assign the issue to you)?

**Comment by meisterfuu** [comment]
I will create an account called "meisterfuu" later. 
Ok, but why does the output say something about "/smack-core/src/main/java/org/jivesoftware/smack/util/ArrayBlockingQueueWithShutdown.java"?

**Comment by Flowdalic** [comment]
Cherry picked with minor modifications as https://github.com/igniterealtime/Smack/commit/05c97c494bf1b5c122c2ef0a7cfdb60a8f3ff828

**Comment by meisterfuu** [comment]
JIRA username is "meisterfuu"

**Pull request closed without merging**